cocos2d/core/base-nodes/CCNodeWebGLRenderCmd.js

```diff
@@ -57,7 +57,7 @@
     // to ease the transition from using glPrograms directly, to 
     // using glProgramStates. 
     Object.defineProperty(proto, '_shaderProgram', {
-        set: function () { this.setShaderProgram.apply(this, arguments); },
```

Comment:
setter will only have one argument, so there is no need to use apply. Also, pass arguments directly to another function call is probably causing memory leak in v8